tests/lint-groups.rs

```diff
@@ -0,0 +1,132 @@
+//! This test checks and updates `book/src/granular_lint_groups.md`
+//!
+//! Specifically the parts in between lint-group-start and lint-group-end are not meant to be edited
+//! by hand and are instead generated based on the const `GROUPS` slice.
+#![feature(rustc_private)]
+
+use std::collections::HashSet;
+use std::{env, fs};
+
+use clippy_lints::LintInfo;
+use clippy_lints::declared_lints::LINTS;
+use indoc::formatdoc;
+use itertools::Itertools;
+use regex::{Captures, Regex};
+
+const GROUPS: &[(&str, &[&str])] = &[
```

Comment:
Did you think about adding an optional "subgroup tag" to the `define_clippy_lints` macro and then use that information to generate those groups, instead of listing them out here?

This might be better in the long run. WDYT?

Reply:
Yeah, that sounds like a better idea. Implemented that for the macro, though not sure if the current syntax is a good idea.
I was initially worried that the term "lint group" was too ambiguous for this because we already use "lint group" and "lint category" interchangeably everywhere which already has its own meaning, but you mentioned "subgroup" which I like more as a name for this, so it uses that consistently now.